src/pages/organize/[orgId]/people/incoming/index.tsx

```diff
@@ -136,7 +84,56 @@ const IncomingPage: PageWithLayout<Props> = ({ orgId }) => {
           </Select>
         </FormControl>
       </Box>
-      <RenderSubmissions />
+      <ZUIFuture
+        future={joinSubmissions}
+        ignoreDataWhileLoading
+        skeleton={<Box />}
```

Comment:
Why use a box here? Removing this line (and providing no value for the `skeleton` property) should, unless I'm mistaken, show some sort of loading indicator, or if you want you could use `<CircularProgress/>`.

```suggestion
        skeleton={<CircularProgress />}
```

Right now, with your current changes, it shows nothing so it's difficult for the user to know that data is pending and they might understandably assume that there are no submissions. In that sense, it doesn't really solve the underlying issue described in #2031.

Reply:
Demoing https://github.com/zetkin/app.zetkin.org/pull/2090/commits/85dd36372914d9625dfc79c4ab83ecfe56402308 here with an artificial 1000ms fetch delay!

![2024-09-28 11 14 28](https://github.com/user-attachments/assets/41550862-8dd7-4da5-b551-d7c4cfcfd3f3)